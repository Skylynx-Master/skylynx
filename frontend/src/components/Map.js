import React from 'react';
import { GoogleMap, withScriptjs, withGoogleMap } from 'react-google-maps';
import '../assets/styles/components/Map.scss';


function Mapping () {
  return (
    <GoogleMap defaultZoom={10} defaultCenter={{ lat: 19.5636957, lng: -99.1928209 }} />
  )
}

const WrappedMap = withScriptjs(withGoogleMap(Mapping))

function Map() {
  return(
    <div className="Map">
      <WrappedMap googleMapURL={"https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places"}
      loadingElement={<div style={{ height: "0%" }} />}
          containerElement={<div style={{ height: "100%" }} />}
          mapElement={<div style={{ height: "100%" }} />} />

    </div>
  )
}

export default Map;