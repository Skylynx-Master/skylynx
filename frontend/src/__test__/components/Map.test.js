import React from "react";
import Map from "../../components/Map";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("Map Component Testing", () => {
  const map = mount(
    <BrowserRouter>
      <Map />
    </BrowserRouter>
  );

  test("Map Render", () => {
    expect(map.length).toEqual(1);
  });

});

describe("Map Snapshot", () => {
  test("Test UI Map component", () => {
    const snapmap = create(
      <BrowserRouter>
        <Map />
      </BrowserRouter>
    );
    expect(snapmap.toJSON()).toMatchSnapshot();
  });
});