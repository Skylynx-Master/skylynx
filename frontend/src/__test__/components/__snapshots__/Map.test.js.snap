// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Snapshot Test UI Map component 1`] = `
<div
  className="Map"
>
  <div
    style={
      Object {
        "height": "0%",
      }
    }
  />
</div>
`;
